### ISSUE TITLE
`RustLsp expandMacro` fails to work in standalone file

### ISSUE THREAD

**Issue by jimzk** [mrcjkb/rustaceanvim#90]
### Neovim version (nvim -v)

v0.9.4

### Operating system/version

macOS

### Output of :checkhealth rustaceanvim

```console
==============================================================================
rustaceanvim: require("rustaceanvim.health").check()

Checking for Lua dependencies ~
- OK [mfussenegger/nvim-dap](https://github.com/mfussenegger/nvim-dap) installed.

Checking external dependencies ~
- OK rust-analyzer: found rust-analyzer 0.3.1756-standalone (e402c494b 2023-12-01)
- OK Cargo: found cargo 1.74.0 (ecb9851af 2023-10-18)
- OK rustc: found rustc 1.74.0 (79e9716c9 2023-11-13)
- WARNING       debug adapter: not found.
  Install [lldb](https://lldb.llvm.org/) for extended capabilities.
  A debug adapter (defaults to: LLDB).
  Required for debugging features.

Checking config ~
- OK No errors found in config.

Checking for conflicting plugins ~
- OK No conflicting plugins detected.
```

### How to reproduce the issue

```rust
fn main() {
    println!("Hello, world!");
}
```

Run `RustLsp expandMacro` on macro `println!`

### Expected behaviour

Expand macro

### Actual behaviour

<img width="1512" alt="image" src="https://github.com/mrcjkb/rustaceanvim/assets/10106636/3969421b-9713-4560-a8ab-d9c9d294d76f">

### The minimal config used to reproduce this issue.

```Lua
N/A
```

**Comment by mrcjkb** [comment]
I've implemented a fix for the error message in #91.

But if you're using a non-project file, I don't think `expandMacro` will work, because afaik, rust-analyzer needs a cargo project to do so.

**Comment by jimzk** [comment]
I found the cause. The issue occurs when `cwd` is outside of git repo directory. Thanks @mrcjkb
